// @metaplex-foundation/mpl-core-oracle-example/dist/src/generated/programs/mplCoreOracleExample.js
"use strict";
/**
 * This code was AUTOGENERATED using the kinobi library.
 * Please DO NOT EDIT THIS FILE, instead use visitors
 * to add features, then rerun kinobi to update it.
 *
 * @see https://github.com/metaplex-foundation/kinobi
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMplCoreOracleExampleProgramId = exports.getMplCoreOracleExampleProgram = exports.createMplCoreOracleExampleProgram = exports.MPL_CORE_ORACLE_EXAMPLE_PROGRAM_ID = void 0;
const errors_1 = require("../errors");
exports.MPL_CORE_ORACLE_EXAMPLE_PROGRAM_ID = '4RZ7RhXeL4oz4kVX5fpRfkNQ3nz1n4eruqBn2AGPQepo';
function createMplCoreOracleExampleProgram() {
    return {
        name: 'mplCoreOracleExample',
        publicKey: exports.MPL_CORE_ORACLE_EXAMPLE_PROGRAM_ID,
        getErrorFromCode(code, cause) {
            return (0, errors_1.getMplCoreOracleExampleErrorFromCode)(code, this, cause);
        },
        getErrorFromName(name, cause) {
            return (0, errors_1.getMplCoreOracleExampleErrorFromName)(name, this, cause);
        },
        isOnCluster() {
            return true;
        },
    };
}
exports.createMplCoreOracleExampleProgram = createMplCoreOracleExampleProgram;
function getMplCoreOracleExampleProgram(context, clusterFilter) {
    return context.programs.get('mplCoreOracleExample', clusterFilter);
}
exports.getMplCoreOracleExampleProgram = getMplCoreOracleExampleProgram;
function getMplCoreOracleExampleProgramId(context, clusterFilter) {
    return context.programs.getPublicKey('mplCoreOracleExample', exports.MPL_CORE_ORACLE_EXAMPLE_PROGRAM_ID, clusterFilter);
}
exports.getMplCoreOracleExampleProgramId = getMplCoreOracleExampleProgramId;
